internal/cli/wait.go

```diff
@@ -0,0 +1,109 @@
+/*
+Adapted from
+https://github.com/kubernetes/kubectl/tree/master/pkg/cmd/wait
+*/
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/cmd/wait"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/tilt-dev/tilt/internal/analytics"
+	engineanalytics "github.com/tilt-dev/tilt/internal/engine/analytics"
+	"github.com/tilt-dev/tilt/pkg/model"
+)
+
+var (
+	waitLong = templates.LongDesc(i18n.T(`
+		Experimental: Wait for a specific condition on one or many resources.
+
+		The command takes multiple resources and waits until the specified condition
+		is seen in the Status field of every given resource.
+
+		A successful message will be printed to stdout indicating when the specified
+    condition has been met. You can use -o option to change to output destination.`))
+
+	waitExample = templates.Examples(i18n.T(`
+		# Wait for the tiltfile to load.
+		tilt wait --for=condition=Ready "uiresource/(Tiltfile)"
+
+		# Wait for a deployment to become ready.
+		tilt wait --for=condition=Ready "uiresource/my-deployment"`))
```

Comment:
Should this be resource instead of deployment?

Might be useful to add an explainer for K8s, e.g. "for Kubernetes, this will wait until the Pod is deployed and passing health checks"

(but also this is fine as-is tbh)

Reply:
done, made this a bit clearer